.github/workflows/deploy.yml

```diff
@@ -0,0 +1,38 @@
+name: Publish Docker image
+
+on:
+  push:
+    branches: ['main']
+  pull_request:
+
+jobs:
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: romyme/legal_assist_api
+      
+      - name: Build and push Docker image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+<<<<<<< HEAD
```

Comment:
conflicts have not been handled properly. you should remove `<<<` and `>>>` symbols and only keep relevant parts